default.nix

```diff
@@ -493,215 +519,252 @@ in {
+                    echo "done"
+                  elif ! echo "$out" | grep "File exists" >/dev/null 2>&1; then
+                    echo "'ip -4 route add "${lib.escapeShellArg route}"' failed: $out"
+                    exit 1
+                  fi
+                '') routes4}
+                ${lib.concatMapStrings (route: ''
+                  echo -n "adding route ${route}... "
+                  if out=$(ip -6 route add ${route} 2>&1 && echo ${lib.escapeShellArg ("ip -6 route del " + route)} >> "$state"); then
+                    echo "done"
+                  elif ! echo "$out" | grep "File exists" >/dev/null 2>&1; then
+                    echo "'ip -6 route add "${lib.escapeShellArg route}"' failed: $out"
+                    exit 1
+                  fi
+                '') routes6}
+              '';
+              preStop = ''
+                state="/run/nixos/network/routes/${interface}"
+                ${lib.optionalString (icfg.bridge != null && !icfg.hostapd.enable) ''
+                  ip link set "${interface}" nomaster up && echo "${interface} " >> "/run/${icfg.bridge.name}.interfaces" || true
+                ''}
+                if [ -e "$state" ]; then
+                  while read cmd; do
+                    echo -n "running route delete command $cmd... "
+                    $cmd >/dev/null 2>&1 && echo "done" || echo "failed"
+                  done < "$state"
+                  rm -f "$state"
+                fi
+
+                state="/run/nixos/network/addresses/${interface}"
+                if [ -e "$state" ]; then
+                  while read cidr; do
+                    echo -n "deleting address $cidr... "
+                    ip addr del "$cidr" dev "${interface}" >/dev/null 2>&1 && echo "done" || echo "failed"
+                  done < "$state"
+                  rm -f "$state"
+                fi
+              '';
+            };
+        })
+    // lib.flip lib.mapAttrs' bridges (interface: value:
+      let
+        escapedInterface = utils.escapeSystemdPath interface;
+      in
+      {
+        name = "${escapedInterface}-netdev";
+        value = router-lib.mkServiceForIf' { inherit interface; includeBasicDeps = false; } {
+          description = "Router Bridge Interface ${interface}";
+          wantedBy = [ "network-setup.service" "network.target" "sys-subsystem-net-devices-${escapedInterface}.device" ];
+          partOf = [ "network-setup.service" ];
+          after = [ "network-pre.target" ]
+            # soft dependency, order it but don't require
+            ++ map router-lib.mainDepForIf value;
+          before = [ "network-setup.service" ];
+          serviceConfig.Type = "oneshot";
+          serviceConfig.RemainAfterExit = true;
+          path = [ pkgs.iproute2 ];
+          script =
+            let
+              vlan_filtering = builtins.any (x: config.router.interfaces.${x}.bridge.vlans != [ ]) value;
```

Comment:
This takes vlans for all bridges, not just this bridge. Fixing what I mentioned above should help fix this as well.